domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaygroup/PageShowConditionValidatorImplTest.java

```diff
@@ -85,11 +92,71 @@ public void returnsNoValidationErrorsOnSuccess() throws InvalidShowConditionExce
         when(mockShowConditionParser.parseShowCondition("someShowCondition"))
                 .thenReturn(validParsedShowCondition);
 
+        CaseTypeEntity caseTypeEntity = new CaseTypeEntity();
+        caseTypeEntity.setReference("SimpleType");
+        displayGroup.setCaseType(caseTypeEntity);
+
         ValidationResult result = testObj.validate(displayGroup, UNUSED_DISPLAY_GROUPS);
 
         assertThat(result.isValid(), is(true));
     }
 
+    @Test
+    public void successfullyValidatesShowConditionForCustomComplexField() throws InvalidShowConditionException {
+        String matchingCaseFieldId = "complexName";
+        String matchingCaseFieldKey = matchingCaseFieldId + ".LastNameWithSomeCplxFields.SomeComplexFieldsCode.AddressUKCode.Country";
+        String showCondition = matchingCaseFieldKey + "=\"UK\"";
+
+        ShowCondition validParsedShowCondition = new ShowCondition.Builder().showConditionExpression(showCondition).field(matchingCaseFieldKey).build();
+        when(mockShowConditionParser.parseShowCondition(showCondition))
+            .thenReturn(validParsedShowCondition);
+
+        displayGroup.setShowCondition(showCondition);
+        displayGroup.setType(DisplayGroupType.PAGE);
+        EventEntity event = mock(EventEntity.class);
+        when(event.hasField(matchingCaseFieldId)).thenReturn(true);
+        displayGroup.setEvent(event);
+        CaseTypeEntity caseTypeEntity = new CaseTypeEntity();
+        caseTypeEntity.setReference("SimpleType");
+        caseTypeEntity.addCaseField(caseFieldEntity("NonMatchingCaseFieldId1"));
+        caseTypeEntity.addCaseField(caseFieldEntity(matchingCaseFieldId, exampleFieldTypeEntityWithComplexFields()));
+        caseTypeEntity.addCaseField(caseFieldEntity("NonMatchingCaseFieldId2"));
+        displayGroup.setCaseType(caseTypeEntity);
+
+        ValidationResult result = testObj.validate(displayGroup, UNUSED_DISPLAY_GROUPS);
+
+        assertThat(result.isValid(), is(true));
+    }
+
+    @Test
+    public void TabShowCondition_shouldAddErrorForInvalidShowConditionForCustomComplexField() throws InvalidShowConditionException {
```

Comment:
Shouldn't this test be inside `TabShowConditionValidatorImplTest`?

Reply:
Name was copied from TabShowConditionValidatorImplTest. It is fixed now.